pandas/io/formats/style.py

```diff
@@ -783,15 +783,29 @@ def _copy(self, deepcopy: bool = False) -> Styler:
             precision=self.precision,
             caption=self.caption,
             uuid=self.uuid,
-            table_styles=self.table_styles,
+            table_attributes=self.table_attributes,
+            cell_ids=self.cell_ids,
             na_rep=self.na_rep,
+            uuid_len=self.uuid_len,
         )
+
+        styler.hidden_index = self.hidden_index
+
         if deepcopy:
             styler.ctx = copy.deepcopy(self.ctx)
             styler._todo = copy.deepcopy(self._todo)
+            styler.table_styles = copy.deepcopy(self.table_styles)
+            styler.hidden_columns = copy.deepcopy(self.hidden_columns)
+            styler.cell_context = copy.deepcopy(self.cell_context)
+            styler.tooltips = copy.deepcopy(self.tooltips)
         else:
             styler.ctx = self.ctx
             styler._todo = self._todo
+            styler.table_styles = self.table_styles
+            styler.hidden_columns = self.hidden_columns
+            styler.cell_context = self.cell_context
+            styler.tooltips = self.tooltips
```

Comment:
`_Tooltips` class doesn't have its own __copy__ and __deepcopy__ methods.

Reply:
Sorry if I don't get it, but does the `_Tooltips` class really needs this methods to achieve the intended behavior? 

```python
class Bar:
    pass

class Foo:
    def __init__(self):
        self.bar = [Bar()]

foo = Foo()
foo_a = foo
foo_c = copy.copy(foo)
foo_dc = copy.deepcopy(foo)

print(id(foo))
print(id(foo_a))
print(id(foo_c))
print(id(foo_dc))
print()
print(id(foo.bar))
print(id(foo_a.bar))
print(id(foo_c.bar))
print(id(foo_dc.bar))
print()
print(id(foo.bar[0]))
print(id(foo_a.bar[0]))
print(id(foo_c.bar[0]))
print(id(foo_dc.bar[0]))
```
-->
```
139949532734368
139949532734368
139949532734464
139949532734272

139949534039680
139949534039680
139949534039680
139949534036352

139949532734752
139949532734752
139949532734752
139949532734128  <== !
```
`copy.deepcopy` does the trick without `Foo.__deepcopy__` method.

Reply:
i think you're right, nice catch.